src/jit/gentree.h

```diff
@@ -574,11 +574,21 @@ struct GenTree
         return isContained() && IsCnsIntOrI() && !isUsedFromSpillTemp();
     }
 
+#if FEATURE_X87_DOUBLES
```

Comment:
Here was one example: https://github.com/dotnet/coreclr/pull/9802/files#diff-c2896a82e1a2272c25849727a769712dL2445 and There were a couple other similar instances scattered around.

It isn't strictly required to support float constants (as we could just check the real type and do the appropriate thing). However, all the upcasting and downcasting is not ideal:
* It means we incur the cost of the upcast/downcast whenver we do it
* It means the values computed for floats could be "off" (it may be possible to compute a value that couldn't be computed without upcasting, resulting in a different end result)